src/planwise/client/scenarios/handlers.cljs

```diff
@@ -152,10 +153,13 @@
  :scenarios/open-changeset-dialog
  in-scenarios
  (fn [db [_ changeset-index]]
-   (assoc db
-          :view-state        :changeset-dialog
-          :view-state-params {:changeset-index changeset-index}
-          :changeset-dialog  (get-in db [:current-scenario :changeset changeset-index]))))
+   (let [last-change (:changeset-dialog db)
+         provider    (get-in db [:current-scenario :changeset changeset-index])
+         condition   (reduce = (map :provider-id [provider last-change]))]
```

Comment:
This looks like a complicated way of saying `(= (:provider-id provider) (:provider-id last-change))`. Also, what is the condition being tested here? And why changing the contents of `changeset-dialog` depend on it?